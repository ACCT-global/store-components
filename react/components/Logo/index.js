import React, { Component } from 'react'
import PropTypes from 'prop-types'

/**
 * Logo of the store
 */
export default class Logo extends Component {
  static propTypes = {
    /** URL of the logo */
    url: PropTypes.string,
    /** Title to be displayed as alt text */
    title: PropTypes.string.isRequired,
    /** Logo's color */
    color: PropTypes.string,
    /** Logo's width */
    width: PropTypes.number,
    /** Logo's height */
    height: PropTypes.number,
    /** Set label visibility */
    showLabel: PropTypes.bool,
  }

  static defaultProps = {
    title: 'VTEX logo',
    color: '#F71963',
    showLabel: true,
    width: 493,
    height: 177,
  }

  render() {
    const { width, height, color, showLabel, url, title } = this.props
    const imgInlineStyles = { maxWidth: width, maxHeight: height }

    if (url) {
      return <img className="store-logo" style={imgInlineStyles} src={url} alt={title} />
    }

    if (!showLabel) {
      return (
        <div className="store-logo" style={{ width, height }}>
          <svg xmlns="http://www.w3.org/2000/svg" width={`${width}px`} height={`${height}px`} viewBox="0 0 24 24">
            <path fill={color} d="M22.2029951,2 L4.28456322,2 C2.89573023,2 2.00548,3.45312422 2.65431247,4.64375547 L4.44783693,7.95552421 L1.19680743,7.95552421 C0.296493196,7.95552421 -0.280470412,8.85287845 0.140086685,9.62472859 L5.90522357,20.1887392 C6.35336788,21.0110278 7.56744234,21.0045158 8.01748106,20.1831745 L9.58332217,17.3210893 L11.5485268,20.925543 C12.2398649,22.1940814 14.1130726,22.195147 14.8071339,20.9280294 L23.7877802,4.5463122 C24.4227598,3.38729378 23.5564264,2 22.2029951,2 M14.1533286,8.98098734 L10.2804619,16.0501827 C9.98102805,16.5965991 9.17318428,16.5957703 8.87505286,16.0487619 L5.03936375,9.00975856 C4.75958436,8.4962574 5.14355582,7.87820895 5.74242344,7.87820895 L13.4699233,7.87820895 C14.0535173,7.87820895 14.427188,8.48110219 14.1533286,8.98098734" className="fill-rebel-pink">
            </path>
            <path d="M0 0h24v24H0z" fill="none"></path>
          </svg>
        </div>
      )
    }

    return (
      <div className="store-logo" style={{ width, height }}>

        <svg width={width} height={height} viewBox='0 0 142 80' fill='none' xmlns='http://www.w3.org/2000/svg'>
          <rect x='12.5' y='12.5' width='127' height='65' stroke='#000' strokeWidth='5' />
          <rect x='2.5' y='2.5' width='127' height='65' fill='#fff' stroke='#000' strokeWidth='5' />
          <path d='M17.0145 32.5042V14H25.1016C27.749 14 29.9326 14.8256 31.6525 16.4767C33.3916 18.1089 34.2612 20.2535 34.2612 22.9105V23.5937C34.2612 26.2887 33.4013 28.4522 31.6814 30.0844C29.9616 31.6976 27.7587 32.5042 25.0726 32.5042H17.0145ZM29.3046 23.1667C29.3046 21.4017 28.9181 20.0447 28.1451 19.0958C27.3915 18.1468 26.29 17.6724 24.8407 17.6724H21.7392V28.8318H24.7828C26.2514 28.8318 27.3722 28.3478 28.1451 27.3799C28.9181 26.412 29.3046 25.0076 29.3046 23.1667Z' fill='#000' />
          <path d='M47.9793 20.0067C47.9793 19.2666 47.7377 18.6877 47.2546 18.2702C46.7715 17.8337 46.0952 17.6154 45.2256 17.6154H42.0082V22.6542H44.9068C45.9116 22.6542 46.6749 22.4265 47.1967 21.971C47.7184 21.5155 47.9793 20.8608 47.9793 20.0067ZM52.646 19.8929V20.3483C52.646 21.582 52.2789 22.6637 51.5446 23.5937C50.8296 24.5236 49.8247 25.1974 48.53 25.6149L53.3997 31.9348V32.5042H48.3271L43.5444 26.0135H42.0082V32.5042H37.2255V14H45.3416C47.5445 14 49.3127 14.5314 50.646 15.5942C51.9794 16.638 52.646 18.0709 52.646 19.8929Z' fill='#000' />
          <path d='M55.5682 32.5042V14H68.6119V17.7862H60.3509V21.2024H67.3655V25.1025H60.3509V28.6894H68.6119V32.5042H55.5682Z' fill='#000' />
          <path d='M80.0442 18.3271H79.9862C79.8896 18.7447 79.7543 19.2476 79.5804 19.8359C79.4065 20.4053 79.1939 21.0316 78.9427 21.7148L77.4934 25.6434H82.537L81.0877 21.7148C80.8365 21.0316 80.6239 20.4053 80.45 19.8359C80.2761 19.2476 80.1408 18.7447 80.0442 18.3271ZM89.9284 32.5042H85.0588L83.8703 29.2873H76.1601L74.9716 32.5042H70.044V31.9917L77.4065 14H82.566L89.9284 31.9917V32.5042Z' fill='#000' />
          <path d='M108.725 23.5368L105.217 32.2195H101.797L98.2605 23.6506C98.0866 23.2141 97.8644 22.5404 97.5939 21.6294C97.3233 20.7184 97.1398 20.0067 97.0431 19.4943H96.9852C97.0431 20.2535 97.0625 21.079 97.0431 21.971C97.0431 22.863 97.0045 23.5557 96.9272 24.0492L95.6518 32.5042H91.072V31.9917L93.9416 14H98.5794L103.594 25.9565H103.652L108.725 14H113.246L116 31.9917V32.5042H111.275L110 23.793C109.903 23.1856 109.836 22.4645 109.797 21.6294C109.778 20.7754 109.787 20.0637 109.826 19.4943H109.768C109.671 20.1016 109.517 20.8133 109.304 21.6294C109.111 22.4455 108.918 23.0813 108.725 23.5368Z' fill='#000' />
          <path d='M16.3768 42.677V42.3923C16.3768 40.6083 17.0242 39.2229 18.3189 38.236C19.6136 37.2302 21.3334 36.7272 23.4784 36.7272C25.6234 36.7272 27.3046 37.1542 28.522 38.0083C29.7587 38.8433 30.4447 40.0675 30.58 41.6806V42.0507H26.0292C25.9712 41.4434 25.7393 40.9784 25.3335 40.6558C24.9277 40.3332 24.3576 40.1718 23.6233 40.1718C22.8697 40.1718 22.2706 40.3237 21.8262 40.6273C21.4011 40.931 21.1885 41.339 21.1885 41.8514C21.1885 42.3449 21.3238 42.7339 21.5943 43.0186C21.8842 43.2843 22.4542 43.5311 23.3045 43.7588C23.5364 43.8347 24.0195 43.977 24.7538 44.1858C25.5074 44.3756 26.1065 44.5369 26.5509 44.6698C27.2466 44.8975 27.836 45.1442 28.3191 45.4099C28.8022 45.6567 29.1983 45.9318 29.5075 46.2355C29.9519 46.672 30.3094 47.2129 30.58 47.8582C30.8505 48.5034 30.9858 49.1487 30.9858 49.794V50.0787C30.9858 51.9576 30.3094 53.4189 28.9568 54.4627C27.6041 55.4876 25.778 56 23.4784 56C21.2175 56 19.4397 55.5445 18.145 54.6335C16.8696 53.7036 16.1546 52.3656 16 50.6196V50.221H20.6668C20.6861 50.9991 20.9469 51.5875 21.4494 51.986C21.9711 52.3846 22.7054 52.5838 23.6523 52.5838C24.4253 52.5838 25.0533 52.413 25.5364 52.0714C26.0388 51.7108 26.29 51.2553 26.29 50.705C26.29 50.1546 26.1451 49.7371 25.8553 49.4524C25.5654 49.1487 25.0533 48.883 24.319 48.6553C24.1644 48.5983 23.1982 48.3137 21.4204 47.8012C19.6426 47.2888 18.4445 46.7289 17.8261 46.1216C17.4203 45.7231 17.0725 45.2012 16.7826 44.5559C16.5121 43.8916 16.3768 43.2653 16.3768 42.677Z' fill='#000' />
          <path d='M37.6637 40.8551H32.3012V37.0688H47.6059V40.8551H42.4464V55.573H37.6637V40.8551Z' fill='#000' />
          <path d='M48.4954 46.6625V45.9793C48.4954 43.2654 49.4326 41.0354 51.307 39.2893C53.2008 37.5243 55.5776 36.6418 58.4376 36.6418C61.2975 36.6418 63.6647 37.5243 65.5392 39.2893C67.4329 41.0354 68.3798 43.2654 68.3798 45.9793V46.6625C68.3798 49.3765 67.4329 51.6159 65.5392 53.381C63.6647 55.127 61.2975 56 58.4376 56C55.5776 56 53.2008 55.127 51.307 53.381C49.4326 51.6159 48.4954 49.3765 48.4954 46.6625ZM58.4086 40.5989C56.882 40.5989 55.6839 41.1208 54.8143 42.1646C53.9448 43.1894 53.51 44.5749 53.51 46.3209C53.51 48.01 53.9544 49.386 54.8433 50.4488C55.7322 51.5116 56.9207 52.043 58.4086 52.043C59.9159 52.043 61.114 51.521 62.0029 50.4772C62.9111 49.4144 63.3652 48.029 63.3652 46.3209C63.3652 44.5749 62.9208 43.1894 62.0319 42.1646C61.1623 41.1208 59.9545 40.5989 58.4086 40.5989Z' fill='#000' />
          <path d='M82.1171 43.0756C82.1171 42.3354 81.8756 41.7565 81.3925 41.339C80.9094 40.9025 80.233 40.6843 79.3635 40.6843H76.146V45.7231H79.0446C80.0495 45.7231 80.8128 45.4953 81.3345 45.0399C81.8563 44.5844 82.1171 43.9296 82.1171 43.0756ZM86.7839 42.9617V43.4172C86.7839 44.6508 86.4167 45.7326 85.6824 46.6625C84.9674 47.5925 83.9626 48.2662 82.6679 48.6837L87.5375 55.0036V55.573H82.465L77.6823 49.0823H76.146V55.573H71.3633V37.0688H79.4794C81.6824 37.0688 83.4505 37.6002 84.7839 38.663C86.1172 39.7069 86.7839 41.1397 86.7839 42.9617Z' fill='#000' />
          <path d='M89.706 55.573V37.0688H102.75V40.8551H94.4887V44.2712H101.503V48.1713H94.4887V51.7583H102.75V55.573H89.706Z' fill='#000' />
          <defs>
            <clipPath id="clip0">
              <rect width="127" height="65" fill="white" />
            </clipPath>
          </defs>
        </svg>
      </div>
    )
  }
}
