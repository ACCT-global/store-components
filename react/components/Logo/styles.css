.maxSizeDesktop {
  max-width: 150px;
  max-height: 75px;
}

.maxSizeMobile {
  max-width: 90px;
  max-height: 40px;
}

.logoContainer {}