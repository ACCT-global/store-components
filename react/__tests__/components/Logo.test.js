import React from 'react'

import Logo from '../../Logo'
import { render } from 'enzyme'

describe('<Logo /> component', () => {
  const renderComponent = customProps => {
    const props = {
      ...customProps,
    }
    const comp = <Logo {...props} />

    return render(comp)
  }

  it('should be rendered', () => {
    const wrapper = renderComponent()
    expect(wrapper).toBeTruthy()
  })

  it('should match snapshot with label', () => {
    const wrapper = renderComponent()
    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot without label', () => {
    const wrapper = renderComponent({ showLabel: false })
    expect(wrapper).toMatchSnapshot()
  })
})
