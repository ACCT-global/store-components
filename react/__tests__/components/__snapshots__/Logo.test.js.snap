// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> component should match snapshot with label 1`] = `
<DocumentFragment>
  <div
    class="store-logo logoContainer sizeDesktop pv4 ph6"
  >
    <svg
      fill="none"
      viewBox="0 0 142 80"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        fill="#F2F4F5"
        height="80"
        stroke="#E3E4E6"
        width="142"
      />
      <path
        d="M84.8156 47.5402L77.8157 33.9407C77.5479 33.4222 77.0282 33.0703 76.4349 33.0108C75.8434 32.9445 75.2572 33.1876 74.8844 33.6381L68.8854 40.9206L64.9707 38.386C64.5717 38.1293 64.08 38.0392 63.6162 38.1412C63.149 38.2432 62.7447 38.5271 62.4997 38.9249L57.2497 47.4246C56.9242 47.9499 56.9172 48.6026 57.227 49.1381C57.5367 49.6719 58.1195 50 58.7495 50H83.2494C83.8567 50 84.4184 49.694 84.7386 49.1925C85.0589 48.691 85.0851 48.0655 84.8156 47.5402Z"
        fill="#CACBCC"
      />
      <path
        d="M67.5 33C69.433 33 71 31.433 71 29.5C71 27.567 69.433 26 67.5 26C65.567 26 64 27.567 64 29.5C64 31.433 65.567 33 67.5 33Z"
        fill="#CACBCC"
      />
      <defs>
        <clippath
          id="clip0"
        >
          <rect
            fill="white"
            height="65"
            width="127"
          />
        </clippath>
      </defs>
    </svg>
  </div>
</DocumentFragment>
`;

exports[`<Logo /> component should match snapshot without label 1`] = `
<DocumentFragment>
  <div
    class="store-logo logoContainer sizeDesktop pv4 ph6"
  >
    <svg
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="fill-rebel-pink"
        d="M22.2029951,2 L4.28456322,2 C2.89573023,2 2.00548,3.45312422 2.65431247,4.64375547 L4.44783693,7.95552421 L1.19680743,7.95552421 C0.296493196,7.95552421 -0.280470412,8.85287845 0.140086685,9.62472859 L5.90522357,20.1887392 C6.35336788,21.0110278 7.56744234,21.0045158 8.01748106,20.1831745 L9.58332217,17.3210893 L11.5485268,20.925543 C12.2398649,22.1940814 14.1130726,22.195147 14.8071339,20.9280294 L23.7877802,4.5463122 C24.4227598,3.38729378 23.5564264,2 22.2029951,2 M14.1533286,8.98098734 L10.2804619,16.0501827 C9.98102805,16.5965991 9.17318428,16.5957703 8.87505286,16.0487619 L5.03936375,9.00975856 C4.75958436,8.4962574 5.14355582,7.87820895 5.74242344,7.87820895 L13.4699233,7.87820895 C14.0535173,7.87820895 14.427188,8.48110219 14.1533286,8.98098734"
        fill="#F71963"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
</DocumentFragment>
`;
